pkg/identityserver/identityserver.go

```diff
@@ -172,12 +175,11 @@ func New(c *component.Component, config *Config) (is *IdentityServer, err error)
 	}
 
 	if is.config.DevEUIBlock.Enabled {
-		err := gormstore.Transact(is.Context(), is.db, func(db *gorm.DB) error {
-			err = gormstore.GetEUIStore(db).CreateEUIBlock(is.Context(), "dev_eui", is.config.DevEUIBlock.Prefix, is.config.DevEUIBlock.InitCounter)
-			if err != nil {
-				return err
-			}
-			return nil
+		if is.config.DevEUIBlock.Prefix.IsZero() {
+			return nil, errNoDevEUIPrefix.New()
+		}
+		err := is.withDatabase(is.Context(), func(db *gorm.DB) error {
+			return gormstore.GetEUIStore(db).InitializeDevEUIBlock(is.Context(), is.config.DevEUIBlock.Prefix, is.config.DevEUIBlock.InitCounter)
```

Comment:
Maybe we should reconsider initializing the block in the database on _every_ IS instance start

Reply:
since configuring new blocks will happen extremely rarely, my suggestion is we move it to `is-db` command.

Reply:
Check https://github.com/TheThingsNetwork/lorawan-stack/pull/5199/commits/05d485f62738a3cc3ac13c2255f8473016dfe77e